frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/stateUpdate/vmSettings/storage-tab-state-update.ts

```diff
@@ -46,11 +59,28 @@ export const prefillInitialDiskUpdater = ({ id, prevState, dispatch, getState }:
         );
       }
     } else {
-      dispatch(
-        vmWizardInternalActions[InternalActionType.UpdateStorage](id, {
-          id: oldSourceStorage ? oldSourceStorage.id : getNextIDResolver(getStorages(state, id))(),
-          ...newSourceStorage,
-        }),
+      if (newType !== oldType) {
+        dispatch(
+          vmWizardInternalActions[InternalActionType.UpdateStorage](id, {
+            id: oldSourceStorage
+              ? oldSourceStorage.id
+              : getNextIDResolver(getStorages(state, id))(),
+            ...newSourceStorage,
+          }),
+        );
+        return;
+      }
+      [
+        VMWizardStorageType.PROVISION_SOURCE_DISK,
+        VMWizardStorageType.WINDOWS_GUEST_TOOLS,
```

Comment:
this does not concern provision storages. Can you please create a separate updater for that? - should be 2nd in order

Reply:
also concerns V2V_VMWARE_IMPORT and V2V_VMWARE_IMPORT_TEMP

Reply:
why it doesn't concern provision storages ? the disk validations are for all storages no ?

Reply:
it concerns only storages created by us in the current instance of the wizard

- other storages are external or under user control (UI) - so it is probably safer to not to meddle in these values